Pages/Loginpage.cs

```diff
@@ -1,41 +1,51 @@
-﻿using OpenQA.Selenium;
+﻿using Marsproject.Support;
+using OpenQA.Selenium;
 using OpenQA.Selenium.Chrome;
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Security.Cryptography.X509Certificates;
 using System.Text;
 using System.Threading.Tasks;
 
 namespace Marsproject.Pages
 {
-    public class Loginpage
-    {
+    public class Loginpage : NewCommondriver
 
-        public By SignInButton = By.XPath("//*[contains(text(),'Sign')]");
+    {
+        public Loginpage(IWebDriver driver)
+        {
+            this.driver = driver;
+        }
+        //  public By SignInButton = By.XPath("//*[contains(text(),'Sign')]");
 
-        public By Email_field = By.Name("email");
+        //public By Email_field = By.Name("email");
 
-        public By Password_field = By.Name("password");
+        //public By Password_field = By.Name("password");
 
-        public By Loginbutton = By.XPath("//*[contains(text(),'Login')]");
+        //public By Loginbutton = By.XPath("//*[contains(text(),'Login')]");
 
         public By homepage_text = By.XPath("//*[contains(text(),'Mars Logo')]");
```

Comment:
change         public By homepage_text = By.XPath("//*[contains(text(),'Mars Logo')]"); 

to 

public IWebElement homepage_text = driver.FindElement(By.XPath("//*[contains(text(),'Mars Logo')]"));

the reason behind this is you doing the unnecessary  find by in the method! think about clean code

Reply:
fix this in all pages